app/controllers/admin/badges_controller.rb

```diff
@@ -6,22 +6,62 @@ def index
       @badges = Badge.all
     end
 
+    def new
+      @badge = Badge.new
+    end
+
+    def edit
+      @badge = Badge.find(params[:id])
+    end
+
+    def create
+      @badge = Badge.new(badge_params)
+
+      if @badge.save
+        flash[:success] = "Badge has been created!"
+        redirect_to admin_badges_path
+      else
+        flash[:danger] = @badge.errors.full_messages.to_sentence
```

Comment:
note for the future, we have a neat shortcut in the base class of all models:

https://github.com/forem/forem/blob/c1fe6a64885efa55e5d467aa0e5e8c0d01aad212/app/models/application_record.rb#L56-L58

this allows you to get the same result with `badge.errors_as_sentence`

Reply:
Thank you so much for pointing this out (I will definitely use it in the future) and for QAing and reviewing my PR @rhymes! I super appreciate it. 😄